contracts/src/TeleporterTokenSource.sol

```diff
@@ -594,12 +681,15 @@ abstract contract TeleporterTokenSource is
             "TeleporterTokenSource: collateral needed for destination"
         );
 
-        // Lock the amount in this contract to be sent.
+        // Deposit the funds sent from the user to the bridge,
+        // and set to adjusted amount after deposit.
         amount = _deposit(amount);
-        require(amount > fee, "TeleporterTokenSource: insufficient amount to cover fees");
+        if (feeAmount > 0) {
+            feeAmount =
+                SafeERC20TransferFrom.safeTransferFrom(IERC20(primaryFeeTokenAddress), feeAmount);
+        }
 
-        // Subtract fee amount from amount prior to scaling.
-        amount -= fee;
+        require(amount > 0, "TeleporterTokenSource: zero amount to send");
```

Comment:
Nit: Can we move this require statement back to directly below `_deposit` since it's intended to check if the value returned there?